### ISSUE TITLE
Ingesting collections larger than 256 KB

### ISSUE THREAD

**Issue by aripollak** [stac-utils/stac-server#107]
Amazon SQS & SNS currently have a payload limit of 256 KB. I have a collection STAC that's about 300 KB, so I'm not able to use the ingestion lambda. Would it make sense to add the capability to pass an S3 URL as the ingestion payload instead of JSON, to be able to ingest a file on S3 that's larger than the SQS payload maximum? This functionality would also make it easier to ingest STACs that are already on S3 if they're accessible by the ingestion lambda.

**Comment by matthewhanson** [comment]
@aripollak Yes! That sounds like a great idea, will get it in the next version

**Comment by marchuffnagle** [comment]
I've created PR #168, which allows ingesting large items by writing them to S3, the sending a message like this to the ingest SNS topic:

```json
{
  "Bucket": "source-bucket",
  "Key": "source-key"
}
```

It occurred to me that we could also use S3 object creation triggers for this. We could create an "ingest" bucket and, when an object is written to the bucket, the ingest Lambda would be automatically triggered to ingest that item. It would essentially act the same way that the SNS topic is now, and would make it a bit easier to use. It would cut out the "write the object's key bucket and key to SNS" step for these large objects.

The new workflow would look something like:

```mermaid
graph LR
smallItem[Small Item]
largeItem[Large Item]
ingestTopic[Ingest SNS Topic]
ingestQueue[Ingest SQS Queue]
subgraph ingestBucket[Ingest S3 Bucket]
  largeItemObject[S3 Object]
end
ingestLambda[Ingest Lambda Function]
elasticsearch[(Elasticsearch)]

smallItem --> ingestTopic
ingestTopic --> ingestQueue
ingestQueue --> ingestLambda
ingestLambda --> elasticsearch

largeItem --> largeItemObject
largeItemObject --> ingestLambda
```

**Comment by matthewhanson** [comment]
So we just need to update the ingest Lambda function to recognize s3 events and a user could 1-way sync this to an s3 bucket. 

We do this in a bit different way when coupling this with [Cirrus](https://github.com/cirrus-geo/cirrus-geo), since the Cirrus outputs to s3 act as the source of truth, and the STAC API is just an index of that, handled through Cirrus publish SNS topics since outputs may be across multiple buckets, and only the STAC JSON files should trigger a STAC API ingest.

However, for a plain STAC API without Cirrus this could be a valuable feature for other users who maintain a separate static Catalog as well.

**Comment by marchuffnagle** [comment]
This was resolved by #179
